larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -257,106 +293,132 @@ void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const
                         firstBadWire = iwire;
-
+                    
                     if (isBadChannel || isLastWire)
                         lastBadWire = iwire;
-
+                    
                     if (isBadChannel && !isLastWire)
                         continue;
 
                     if ((firstBadWire < 0) || (lastBadWire < 0))
                         continue;
-
+                    
                     double firstXYZ[3], lastXYZ[3];
                     theGeometry->Cryostat(icstat).TPC(itpc).Plane(iplane).Wire(firstBadWire).GetCenter(firstXYZ);
                     theGeometry->Cryostat(icstat).TPC(itpc).Plane(iplane).Wire(lastBadWire).GetCenter(lastXYZ);
-
+                    
                     firstBadWire = -1; lastBadWire = -1;
-
+                    
                     PandoraApi::Geometry::LineGap::Parameters parameters;
-
+                    
                     try
                     {
                         parameters.m_lineStartX = -std::numeric_limits<float>::max();
                         parameters.m_lineEndX = std::numeric_limits<float>::max();
-
+                        
                         const unsigned int volumeId(LArPandoraGeometry::GetVolumeID(driftVolumeMap, icstat, itpc));
                         LArDriftVolumeMap::const_iterator volumeIter(driftVolumeMap.find(volumeId));
-
+                        
                         if (driftVolumeMap.end() != volumeIter)
                         {
                             parameters.m_lineStartX = volumeIter->second.GetCenterX() - 0.5f * volumeIter->second.GetWidthX();
                             parameters.m_lineEndX = volumeIter->second.GetCenterX() + 0.5f * volumeIter->second.GetWidthX();
                         }
-
-                        const geo::View_t iview = (geo::View_t)iplane;
+                        
+                        const geo::View_t iview = (geo::View_t)plane.View();
                         const geo::View_t pandoraView(LArPandoraGeometry::GetGlobalView(icstat, itpc, iview));
-
-                        if (pandoraView == geo::kW || pandoraView == geo::kY)
+                        
+                        if(isDualPhase)
                         {
-                            const float firstW(firstXYZ[2]);
-                            const float lastW(lastXYZ[2]);
-
-                            parameters.m_lineGapType = pandora::TPC_WIRE_GAP_VIEW_W;
-                            parameters.m_lineStartZ = std::min(firstW, lastW) - halfWirePitch;
-                            parameters.m_lineEndZ = std::max(firstW, lastW) + halfWirePitch;
+                            if (pandoraView == geo::kW || pandoraView == geo::kZ)
+                            {
+                                const float firstW(firstXYZ[2]);
+                                const float lastW(lastXYZ[2]);
+                                
+                                parameters.m_lineGapType = pandora::TPC_WIRE_GAP_VIEW_U;
```

Comment:
How come the gap type is `U` when the view is either `W` or `Z`

Reply:
It comes from what has been defined in line 112 of LArPandoraInput.cxx